vdl_tools/scrape_enrich/prepare_candid.py

```diff
@@ -1,52 +1,75 @@
 import pandas as pd
+import re
 import vdl_tools.scrape_enrich.prepare_candid_old as cd  # from scrape-enrich
 import vdl_tools.shared_tools.project_config as pc
 import vdl_tools.shared_tools.common_functions as cf  # from common directory: commonly used functions
-
+from vdl_tools.scrape_enrich.fix_candid_funders_txt import clean_csv_file
 paths = pc.get_paths()
 
 # %% 
-def __add_funding_by_year(df_cd: pd.DataFrame):
-    # TODO: note this is is temp fix - added funding totals
-    print ("\nAdding total funding by year")
-    funding_cols = ['total_funding_2017',
-                    'total_funding_2018',
-                    'total_funding_2019',
-                    'total_funding_2020',
-                    'total_funding_2021',
-                    'total_funding_2022',
-                    'total_funding_2023']
-    # checks if funding cols are present
-    df_check = pd.read_excel(paths['candid_source_data']/"candid_main_programs_w_yrs.xlsx", sheet_name='main',  engine='openpyxl' )
-    existing_funding_cols = [col for col in funding_cols if col in df_check.columns]
+# def __add_funding_by_year(df_cd: pd.DataFrame):
+#     # TODO: note this is is temp fix - added funding totals
+#     print ("\nAdding total funding by year")
+#     funding_cols = ['total_funding_2017',
+#                     'total_funding_2018',
+#                     'total_funding_2019',
+#                     'total_funding_2020',
+#                     'total_funding_2021',
+#                     'total_funding_2022',
+#                     'total_funding_2023']
+#     # checks if funding cols are present
+#     df_check = pd.read_excel(paths['candid_source_data']/"candid_main_programs_w_yrs.xlsx", sheet_name='main',  engine='openpyxl' )
+#     existing_funding_cols = [col for col in funding_cols if col in df_check.columns]
+#
+#     if not existing_funding_cols:
+#         print("No funding columns found in the data. Skipping merge.")
+#         return df_cd
+#     keep_cols = ['ein'] + existing_funding_cols
+#     df_cd_by_yr = pd.read_excel(paths['candid_source_data'] / "candid_main_programs_w_yrs.xlsx", sheet_name='main')[
+#         keep_cols]
+#
+#     df_cd_by_yr.columns = ['id'] + ['Funding_'+ x.split("_")[-1] for x in funding_cols]
+#     df_cd_by_yr.fillna(0, inplace=True)
+#     df_cd = df_cd.merge(df_cd_by_yr, on='id', how='left')
+#     return df_cd
 
-    if not existing_funding_cols:
+def __add_funding_by_year(df_cd: pd.DataFrame, start_year=2017, year_col_prefix='total_funding_'):
+    print("\nAdding total funding by year")
+    # Read the columns from the Excel file
+    df_check = pd.read_excel(paths['candid_source_data'] / "candid_main_programs_w_yrs.xlsx", sheet_name='main', engine='openpyxl')
+    # Find columns matching the pattern and >= start_year
+    funding_cols = [col for col in df_check.columns
+                    if re.match(f"{year_col_prefix}\\d{{4}}", col) and int(col[-4:]) >= start_year]
```

Comment:
Something like this to simplify and remove regex
```suggestion
    funding_cols = [col for col in df_check.columns
                    if col.startswith('year_col_prefix') and int(col[-4:]) >= start_year]
```